contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3012,18 +3012,43 @@ static void bbf_ProcessUtility(PlannedStmt *pstmt,
 			}
 			break;
 		case T_RenameStmt:
+		{
+			if (sql_dialect == SQL_DIALECT_TSQL)
+			{
+				RenameStmt *stmt = (RenameStmt *) parsetree;
+
+				PG_TRY();
+				{
+					if (prev_ProcessUtility)
+						prev_ProcessUtility(pstmt, queryString, readOnlyTree, context,
+											params, queryEnv, dest, qc);
+					else
+						standard_ProcessUtility(pstmt, queryString, readOnlyTree, context,
+												params, queryEnv, dest, qc);
+					rename_update_bbf_catalog(stmt);
```

Comment:
you could simply leave this function call under TSQL dialect and remove the rest. 
Since the only diff with non-TSQL case is whethere this function is called or not. 
You could even check dialect inside of this function just like check_extra_schema_restrictions